branch.go

```diff
@@ -65,8 +62,20 @@ func (i *BranchIterator) Free() {
 	C.git_branch_iterator_free(i.ptr)
 }
 
-func (repo *Repository) NewBranchIterator(flags BranchType) (*BranchIterator, error) {
+func (i *BranchIterator) ForEach(f BranchIteratorFunc) error {
+	b, t, err := i.Next()
+
+	for b != nil && err == nil {
```

Comment:
`b` shouldn't ever be `nil` in a situation that you can't decide by checking if `err != nil`, I don't see what that check adds here.